src/spec-configuration/containerCollectionsOCIPush.ts

```diff
@@ -383,15 +383,16 @@ export async function calculateManifestAndContentDigest(output: Log, ociRef: OCI
 	// A canonical manifest digest is the sha256 hash of the JSON representation of the manifest, without the signature content.
 	// See: https://docs.docker.com/registry/spec/api/#content-digests
 	// Below is an example of a serialized manifest that should resolve to '9726054859c13377c4c3c3c73d15065de59d0c25d61d5652576c0125f2ea8ed3'
-	// {"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","config":{"mediaType":"application/vnd.devcontainers","digest":"sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855","size":0},"layers":[{"mediaType":"application/vnd.devcontainers.layer.v1+tar","digest":"sha256:b2006e7647191f7b47222ae48df049c6e21a4c5a04acfad0c4ef614d819de4c5","size":15872,"annotations":{"org.opencontainers.image.title":"go.tgz"}}]}
+	// {"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","artifactType":"application/vnd.devcontainers","config":{"mediaType":"application/vnd.oci.empty.v1+json","digest":"sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a","size":2},"layers":[{"mediaType":"application/vnd.devcontainers.layer.v1+tar","digest":"sha256:b2006e7647191f7b47222ae48df049c6e21a4c5a04acfad0c4ef614d819de4c5","size":15872,"annotations":{"org.opencontainers.image.title":"go.tgz"}}]}
 
 	let manifest: OCIManifest = {
 		schemaVersion: 2,
 		mediaType: 'application/vnd.oci.image.manifest.v1+json',
+		artifactType: DEVCONTAINER_MANIFEST_MEDIATYPE,
 		config: {
-			mediaType: 'application/vnd.devcontainers',
-			digest: 'sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', // A zero byte digest for the devcontainer mediaType.
-			size: 0
+			mediaType: OCI_EMPTY_DESCRIPTOR.mediaType,
```

Comment:
I don't think we should be updating the mediaType from `application/vnd.devcontainers` to `application/vnd.oci.empty.v1+json` to fix this issue.

Reply:
However, I do not have an environment to test registries other than GitHub Registry and AWS ECR, so I cannot determine whether non-compliance with the OCI Image Manifest specification causes problems in other registries.

Reply:
As you pointed out, at least AWS ECR did not need to use an empty descriptor to comply with the OCI image manifest specification. Therefore, I simply changed the content to an empty object, minimizing the scope of the change.

Reply:
@samruddhikhandale 
Could you please re-review the PR?

Reply:
Looping @joshspicer for another set of eyes.